components/ai_chat/ai_chat_tab_helper.cc

```diff
@@ -149,6 +149,25 @@ void AIChatTabHelper::AddToConversationHistory(const ConversationTurn& turn) {
   }
 }
 
+void AIChatTabHelper::UpdateLastEntryInConversationHistory(
```

Comment:
How about `UpdateOrCreateLastAssistantEntry`? Alternatively we could create a simple incrementing `int id` field for each conversation entry, and simply pass that through with the callback so that we know we're updating the correct entry, or handling when the conversation was cleared whilst an old request is in progress (e.g. due to navigation).

Reply:
Done